### PR TITLE
feat(widget-v2): list item button

### PR DESCRIPTION
**Pull request by mikasackermn**

### CODE REVIEW

**Review by nikaaru — APPROVED** [review]

File: widget/embedded/src/components/BlockchainList/BlockchainList.styles.ts [hunk 1]
```diff
@@ -1,9 +1,14 @@
 import { styled } from '@rango-dev/ui';
 
 export const List = styled('ul', {
+  height: 355,
```
Comment (mojtabast) [inline widget/embedded/src/components/BlockchainList/BlockchainList.styles.ts:4]: You are adding a fixed height to a general component. I don't think that's a good idea.

File: widget/embedded/src/components/BlockchainList/BlockchainList.tsx [hunk 2]
```diff
@@ -40,24 +40,20 @@ export function BlockchainList(props: PropTypes) {
...
-            }
-            id={item.chainId as string}
-          />
-        ))}
-      </List>
-    );
+    return blockchains.map((item) => (
+      <ListItemButton
+        key={`${item.name}-${item.chainId}`}
+        hasDivider
```
Comment (mojtabast) [inline widget/embedded/src/components/BlockchainList/BlockchainList.tsx:46]: it says it doesn't exists in props.

File: widget/embedded/src/components/BlockchainList/BlockchainList.tsx [hunk 3]
```diff
@@ -40,24 +40,20 @@ export function BlockchainList(props: PropTypes) {
...
+        onClick={() => onChange(item)}
+        start={<Image src={item.logo} size={30} />}
+        title={
+          <Typography variant="title" size="medium">
+            {item.displayName}
+          </Typography>
+        }
+        id={item.chainId as string}
```
Comment (mojtabast) [inline widget/embedded/src/components/BlockchainList/BlockchainList.tsx:54]: Why we are using type assertion here?

File: widget/ui/src/components/ListItem/ListItem.styles.ts [hunk 4]
```diff
@@ -43,5 +43,9 @@ export const BaseListItem = styled('li', {
         borderBottomLeftRadius: 0,
       },
     },
+    selected: {
```
Comment (mojtabast) [inline widget/ui/src/components/ListItem/ListItem.styles.ts:46]: Why you moved the `selected` from `ListItemButton` to `ListItem`?

File: widget/ui/src/components/ListItemButton/ListItemButton.tsx [hunk 5]
```diff
@@ -2,24 +2,16 @@ import type { ListItemButtonProps } from './ListItemButton.types';
...
 function ListItemButton(props: ListItemButtonProps) {
-  const { onClick, id, style, hasDivider, selected, ...restProps } = props;
-  const onClickWithKey = () => {
-    if (onClick) {
-      onClick(id);
-    }
-  };
+  const { onClick, id, ...restProps } = props;
+
   return (
```
Comment (mojtabast) [inline widget/ui/src/components/ListItemButton/ListItemButton.tsx:12]: The whole idea here was `ListItemButton` is a specific type of `ListItem`, But here and in `token/chains problem` PR, the component has changed and it includes `ListItem` as a part of this component. I mean ListItemButton and ListItem should be used at the same, One of them should do the work.

File: widget/embedded/src/components/BlockchainList/BlockchainList.styles.ts (hunk 1 above)
Comment (yeager-eren) [inline widget/embedded/src/components/BlockchainList/BlockchainList.styles.ts:4]: You are adding a fixed height to a general component. I don't think that's a good idea

File: widget/embedded/src/components/BlockchainList/BlockchainList.tsx (hunk 2 above)
Comment (yeager-eren) [inline widget/embedded/src/components/BlockchainList/BlockchainList.tsx:46]: it says it doesn't exists in props.

File: widget/embedded/src/components/BlockchainList/BlockchainList.tsx (hunk 3 above)
Comment (yeager-eren) [inline widget/embedded/src/components/BlockchainList/BlockchainList.tsx:54]: Why we are using type assertion here?

File: widget/ui/src/components/ListItem/ListItem.styles.ts (hunk 4 above)
Comment (yeager-eren) [inline widget/ui/src/components/ListItem/ListItem.styles.ts:46]: Why you moved the selected from ListItemButton to ListItem?

File: widget/ui/src/components/ListItemButton/ListItemButton.tsx (hunk 5 above)
Comment (yeager-eren) [inline widget/ui/src/components/ListItemButton/ListItemButton.tsx:12]: The whole idea here was ListItemButton is a specific type of ListItem, But here and in token/chains problem PR, the component has changed and it includes ListItem as a part of this component. I mean ListItemButton and ListItem should be used at the same, One of them should do the work.

File: widget/embedded/src/components/BlockchainList/BlockchainList.styles.ts (hunk 1 above)
Comment (mikasackermn) [reply widget/embedded/src/components/BlockchainList/BlockchainList.styles.ts:4]: It's not a general component, this is only used for the blockchain list

**Comment by mikasackermn** [comment]
I returned the listItemButton to the previous state and solved problem with tabIndex so that we can move between the items with the tab, and add onKeyUp for select item with enter key.
Places where it's used:
 1. Tokens list
 2. Blockchains list
 3. Setting page
 4. LiquiditySource page

**Pull request closed without merging**